### ISSUE TITLE
[dv] Enable running OTTF-launched tests with both test and mask ROMs

### ISSUE THREAD

**Issue by timothytrippel** [lowRISC/opentitan#10498]
# Overview
Currently, all OTTF-launched tests (e.g., chip-level tests and, future, mask ROM E2E tests) are only run via the test ROM. However, we need to run said tests with mask ROM as well. To facilitate this, several problems must be addressed:

# Problems
1. The dvsim environment is not configured to be able to load different ROM images into the simulated ROM, and signed binaries into flash. 

2. The mask ROM expects flash images to contain a manifest (with valid signature) at a fixed address, and neither the `test_rom_ext.ld` linker script, nor the `test_rom_ext_start.asm` setup a manifest section in OTTF test images. 

3. The `test_rom_ext.ld` script places all test code in a virtual address space, and relies on the test ROM to configure [Ibex address translation features](https://docs.opentitan.org/hw/ip/rv_core_ibex/doc/#Reg_ibus_addr_matching_0), while the mask ROM does not configure Ibex address translation.

4. The mask ROM locks out (using the Ibex PMP features) the backdoor DV logging and test status reporting mechanism. This needs to be reopened by the OTTF, if a test is loaded with the mask ROM.

To address these problem, the following tasks must be completed.

# Task Burndown:
- [x] extend dvsim configurations and testbench to enable (#10394):
    - [x] loading mask ROM into the simulated ROM
    - [x] loading signed SW binaries into flash (using ROM_EXT image signer tool)
- [x] extend/refactor OTTF to:
    - [x] consolidate test ROM_EXT into OTTF startup stage (the current "test ROM_EXT" stage configures OTTF ISRs so will need to be run right before OTTF launch regardless of what boot stage a test, and therefore the OTTF, is launched at) (#10563)
    - [x] rename OTTF entry point from `_start` to `_ottf_start` (#10666)
    - [x] refactor OTTF linker script to include section for manifest (should look like `rom_ext_common.ld`) (#10666)
    - [x] move OTTF test images to `eflash` region from `eflash_virtual` region so they can be loaded in ROM_EXT slot A by mask ROM for the time being (to unblock mask ROM E2E testing) (#10715)
    - [x] #10863
- [x] refactor meson build rules for signing test images (that include OTTF entry point) (#10715)

**Comment by timothytrippel** [comment]
CC: @moidx @alphan @sriyerg

Just filing this issue to track the tasks needed to complete this.

**Comment by timothytrippel** [comment]
Moved last remaining task into a separate issue.
